### PR TITLE
fix(Core/Spell): Earthliving weapon proc

### PR DESCRIPTION
**Pull request by Kitzunu**
## CHANGES PROPOSED:
-  Backported from https://github.com/TrinityCore/TrinityCore/commit/d685c7079d

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3865

## TESTS PERFORMED:
- Builds
- Tested in-game

## HOW TO TEST THE CHANGES:

Check the issue

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Comment by GMKyle** [comment]
I think you also should remove code from Unit.cpp as well :P
https://github.com/azerothcore/azerothcore-wotlk/blob/ae023831f157949cf12d5da574da68c66029f6d1/src/server/game/Entities/Unit/Unit.cpp#L16490-L16499

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -16487,16 +16487,6 @@ bool Unit::IsTriggeredAtSpellProcEvent(Unit* victim, Aura* aura, SpellInfo const
...
-        case SPELLFAMILY_SHAMAN:
-            {
-                // Blessing of the Eternals, Earthliving proc
-                if (spellProto->SpellIconID == 1929)
-                {
-                    if (victim && (float(victim->GetHealth() * 100.0f / victim->GetMaxHealth()) > 35.0f))
-                        return roll_chance_f(chance);
-                }
-                break;
-            }
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Unit/Unit.cpp:16499]: less stuff in the core and more to the scripts, I like it 👍

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 2]
```diff
@@ -796,6 +797,52 @@ class spell_sha_earthen_power : public SpellScriptLoader
...
+    {
+        PrepareAuraScript(spell_sha_earthliving_weapon_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_BLESSING_OF_THE_ETERNALS_R1))
+            {
+                return false;
+            }
+            return true;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_shaman.cpp:816]: you can return just `sSpellMgr->GetSpellInfo(SPELL_SHAMAN_BLESSING_OF_THE_ETERNALS_R1)` casted to bool

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 3]
```diff
@@ -796,6 +797,52 @@ class spell_sha_earthen_power : public SpellScriptLoader
...
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            int32 chance = 20;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_shaman.cpp:821]: you can use `auto` here

File: src/server/scripts/Spells/spell_shaman.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/scripts/Spells/spell_shaman.cpp:816]: yea but isnt this better for readability?

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
There are many error logs after the server starts

**Comment by FrancescoBorzi** [comment]
@pklloveyou can you please show us the errors that you get?

**Comment by pklloveyou** [comment]
Strange, there is no error log this time, it may be my reason

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**